src/controllers/userController.js

```diff
@@ -85,14 +76,56 @@ const login = async (req, res) => {
 
   const data = {
     user: {
+      id: foundUser.id,
       email: foundUser.email,
       token: authHelper.generateToken({ id: foundUser.id }),
-      username: foundUser.username,
       bio: foundUser.bio
     }
   };
   return successResponse(res, 200, data);
 };
 
+/**
+ * @description updates the profile of a user
+ * @param {object} request express request object
+ * @param {object} response express response object
+ * @param {object} next express next argument
+ * @returns {json} json
+ */
+const editProfile = async (request, response) => {
+  const { body, userData } = request;
+  const {
+    email, firstName, lastName, avatarUrl
+  } = body;
+  if (!email && !firstName && !lastName && !avatarUrl) {
```

Comment:
Your implementation here seems like a user has to update their avatarUrl every time they make a change. Can you explain why? Also, is it possible to abstract the check for the emails and the rest and add it into the already existing validators?

Reply:
Noted, I initially wanted to stop all field entries that would pass the validators because firstName, lastName, email and avatarUrl are all optional, if the user entered another parameter like bookName it would pass the validator but looking at your feedback, I would be able to work on it using another approach as requested.